src/Chapter2.hs

```diff
@@ -491,7 +495,9 @@ True
 >>> isThird42 [42, 42, 0, 42]
 False
 -}
-isThird42 = error "isThird42: Not implemented!"
+isThird42 :: (Eq a, Num a) => [a] -> Bool
+isThird42 (x:y:z:xs) = z == 42
+isThird42 _ = False
```

Comment:
Excellent job on specifying the most general type for this function! :clap: 
Also, it looks like you can improve the implementation by pattern matching on `z` as on number `42` directly in the pattern `(x:y:z:xs)`, instead of comparing the number separately.